l2/storage.go

```diff
@@ -0,0 +1,41 @@
+package l2
+
+import (
+	"os"
+	"path/filepath"
+)
+
+type storage interface {
+	Put(path, name string) error
+	Get(name, path string) error
```

Comment:
I think we should be able to Put/Get a fraction of a file instead of the whole file.

Reply:
This is an optimization that can be adopted in later PR.